### DISCUSSION TITLE
How can I refresh all families of an always-alive provider manually in riverpod?

### DISCUSSION THREAD

**Discussion by RemonAhammad** [rrousselGit/riverpod#1628] [Q&A]
I have used family to cache all data.

```
final provider =
    StateNotifierProvider.family((ref, data) => ListNotifier(ref, data));
```

I want to refresh all families of these providers.

my provider receives an object and returns the same type of object. An object has a generic type of list which is the same type of object. I update every single thing in my nested objects. but, now I have to need to refresh all family providers.

**Comment by rrousselGit** [comment]
It's unreleased, but on master you can do `ref.invalidate(family)`

**Reply by RemonAhammad** [reply]
this error is showing. when I use it. 
```
The argument type 'StateNotifierProviderFamily<SelectedCategoryNotifier, CategoryForAdmin?, CategoryForAdmin>' can't be assigned to the parameter type 'ProviderBase<Object?>'.
```
where I invoke it I haven't an argument. but my provider is not nullable. 
```
ref.invalidate(selectedConfig);
```

**Reply by rrousselGit** [reply]
Like I said, it's unreleased
